pandas/tests/reshape/merge/test_merge.py

```diff
@@ -1442,3 +1442,20 @@ def test_merge_series(on, left_on, right_on, left_index, right_index, nm):
         with pytest.raises(ValueError, match='a Series without a name'):
             result = pd.merge(a, b, on=on, left_on=left_on, right_on=right_on,
                               left_index=left_index, right_index=right_index)
+
+
+@pytest.mark.parametrize("col1, col2, suffixes, expected_cols", [
+    (0, 0, ("", "_dup"), ["0", "0_dup"]),
+    (0, 0, (None, "_dup"), ["0", "0_dup"]),
+    (0, 0, ("_x", "_y"), ["0_x", "0_y"]),
+    ("a", 0, (None, "_y"), ["a", 0])
+])
+def test_merge_suffix(col1, col2, suffixes, expected_cols):
+    a = pd.DataFrame({col1: [1, 2, 3]})
+    b = pd.DataFrame({col2: [4, 5, 6]})
+
+    df = a.merge(b, left_index=True, right_index=True, suffixes=suffixes)
+    assert df.columns.tolist() == expected_cols
+
+    with pytest.raises(ValueError):
```

Comment:
This should be a separate test, `test_merge_suffix_errors`.  Since the test is parametrized, this same check is getting run for each parameter.  Might also want to parametrize the new test for different blank variants, e.g. `(None, '')`, `('', None)`, `('', '')`.  And maybe dupe non-blanks if we decide to go that route.

Also can you check the error message via the `match` parameter, i.e. `pytest.raises(ValueError, match=...)`?

Reply:
done! thanks for the feedback!